src/transformers/integrations/peft.py

```diff
@@ -245,20 +245,27 @@ def add_adapter(self, adapter_config, adapter_name: Optional[str] = None) -> Non
 
         self.set_adapter(adapter_name)
 
-    def set_adapter(self, adapter_name: str) -> None:
+    def set_adapter(self, adapter_name: Union[List[str], str]) -> None:
```

Comment:
Let's do the same in diffusers :-) https://github.com/huggingface/diffusers/pull/5151/files#r1338774571